src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -119,22 +119,24 @@ internal TypeWithAnnotations AsNotAnnotated()
             return Create(Type, NullableAnnotation.NotAnnotated, CustomModifiers);
         }
 
-        internal bool IsPossiblyNullableTypeTypeParameter()
-        {
-            return NullableAnnotation.IsNotAnnotated() &&
-                (Type?.IsPossiblyNullableReferenceTypeTypeParameter() == true ||
-                 Type?.IsNullableTypeOrTypeParameter() == true);
-        }
-
         internal NullableAnnotation GetValueNullableAnnotation()
         {
-            if (IsPossiblyNullableTypeTypeParameter())
+            if (NullableAnnotation.IsAnnotated())
+            {
+                return NullableAnnotation;
+            }
+
+            if (Type is null)
+            {
+                return NullableAnnotation.NotAnnotated;
```

Comment:
Why do we prefer `NotAnnotated` here over `Oblivious`?

Reply:
I've removed the explicit `Type is null` case so the behavior should match the current behavior which returns `this.NullableAnnotation`. That said, I suspect `NullableAnnotation` is typically `NotAnnotated` when `Type is null` since that is the `default` value for the `struct`.

---
In reply to: [456081013](https://github.com/dotnet/roslyn/pull/45993#discussion_r456081013) [](ancestors = 456081013)